### PR TITLE
feat: add storage emulator configuration

### LINKED ISSUE #2794: Storage Emulator feature request.
Recently Firebase Tools added support for firebase Storage emulator.
AngularFire should keep up to date and support this emulator.

Implementation would be very simple as far as I can tell. Just change `src/storage/storage.ts` to the following:
```
import { Inject, Injectable, InjectionToken, NgZone, Optional, PLATFORM_ID } from '@angular/core';
import { createStorageRef } from './ref';
import { Observable } from 'rxjs';
import {
  FIREBASE_APP_NAME,
  FIREBASE_OPTIONS,
  FirebaseAppConfig,
  FirebaseOptions,
  ɵAngularFireSchedulers,
  ɵfetchInstance,
  ɵfirebaseAppFactory,
  ɵkeepUnstableUntilFirstFactory
} from '@angular/fire';
import { UploadMetadata } from './interfaces';
import 'firebase/storage';
import firebase from 'firebase/app';

type UseEmulatorArguments = [string, number];
export const USE_EMULATOR = new InjectionToken<UseEmulatorArguments>('angularfire2.storage.use-emulator');

export const BUCKET = new InjectionToken<string>('angularfire2.storageBucket');
export const MAX_UPLOAD_RETRY_TIME = new InjectionToken<number>('angularfire2.storage.maxUploadRetryTime');
export const MAX_OPERATION_RETRY_TIME = new InjectionToken<number>('angularfire2.storage.maxOperationRetryTime');

/**
 * AngularFireStorage Service
 *
 * This service is the main entry point for this feature module. It provides
 * an API for uploading and downloading binary files from Cloud Storage for
 * Firebase.
 */
@Injectable({
  providedIn: 'any'
})
export class AngularFireStorage {
  public readonly storage: firebase.storage.Storage;

  public readonly keepUnstableUntilFirst: <T>(obs: Observable<T>) => Observable<T>;
  public readonly schedulers: ɵAngularFireSchedulers;

  constructor(
    @Inject(FIREBASE_OPTIONS) options: FirebaseOptions,
    @Optional() @Inject(FIREBASE_APP_NAME) nameOrConfig: string | FirebaseAppConfig | null | undefined,
    @Optional() @Inject(BUCKET) storageBucket: string | null,
    // tslint:disable-next-line:ban-types
    @Inject(PLATFORM_ID) platformId: Object,
    zone: NgZone,
    @Optional() @Inject(USE_EMULATOR) _useEmulator: any, // can't use the tuple here
    @Optional() @Inject(MAX_UPLOAD_RETRY_TIME) maxUploadRetryTime: number | any,
    @Optional() @Inject(MAX_OPERATION_RETRY_TIME) maxOperationRetryTime: number | any,
  ) {
    this.schedulers = new ɵAngularFireSchedulers(zone);
    this.keepUnstableUntilFirst = ɵkeepUnstableUntilFirstFactory(this.schedulers);
    const app = ɵfirebaseAppFactory(options, zone, nameOrConfig);
    const useEmulator: UseEmulatorArguments | null = _useEmulator;

    this.storage = ɵfetchInstance(`${app.name}.storage.${storageBucket}`, 'AngularFireStorage', app, () => {
      const storage = zone.runOutsideAngular(() => app.storage(storageBucket || undefined));
      if(useEmulator) {
        (firebase.storage() as any).useEmulator(...useEmulator); // line 439 in scripts/storage-emulator-integration/tests.ts of https://github.com/firebase/firebase-tools/compare/v9.7.0...v9.8.0#diff-d3f7ed92e4a8a777c34e51979c8092cb1b3039ecdfaf49c2237729f08e8d5a14R439
      }
      
      if (maxUploadRetryTime) {
        storage.setMaxUploadRetryTime(maxUploadRetryTime);
      }
      if (maxOperationRetryTime) {
        storage.setMaxOperationRetryTime(maxOperationRetryTime);
      }
      return storage;
    }, [maxUploadRetryTime, maxOperationRetryTime, useEmulator]);
  }

  ref(path: string) {
    return createStorageRef(this.storage.ref(path), this.schedulers, this.keepUnstableUntilFirst);
  }

  refFromURL(path: string) {
    return createStorageRef(this.storage.refFromURL(path), this.schedulers, this.keepUnstableUntilFirst);
  }

  upload(path: string, data: any, metadata?: UploadMetadata) {
    const storageRef = this.storage.ref(path);
    const ref = createStorageRef(storageRef, this.schedulers, this.keepUnstableUntilFirst);
    return ref.put(data, metadata);
  }

}

```

Of course we need to update the docs as well. (`sample/src/app/app.module.ts`, `site/src/get-started/local-development.md`, `docs/emulators/emulators.md`)
I'd love to make a pull request for this, but it seems like my account is not authorized to push to a new branch for some reason. @jamesdaniels

### PR DESCRIPTION
**Pull request by hmmmmniek**
add support for enabling storage emulator, since this was recently added
to the firebase web sdk

Fixes #2794

### Checklist

   - Issue number for this PR: #2794
   - Docs included?: yes 
   - Test units included?: is this necessary in this case? and if so, where can I add the tests? I can't find any existing specific emulator related tests for firestore for example. Would prefer to work of an example.
   - In a clean directory, `yarn install`, `yarn test` run successfully? yes
### Description

The storage emulator is being worked on and now in early access.
When using the cli command `firebase --open-sesame storageemulator` we are already able to try it out.
Next step is to add support for this new emulator in this project as well!
So that is what this is, a tiny pull request to call `storage.useEmulator(..)` just like how it is already done for the firestore emulator, functions emulator etc. etc. 

### Code sample
examples can be found in 
`sample/src/app/app.module.ts`
`site/src/get-started/local-development.md`
`docs/emulators/emulators.md`

### CODE REVIEW

File: site/src/get-started/local-development.md [hunk 1]
```diff
@@ -9,12 +9,13 @@ eleventyNavigation:
...
 In this guide, we'll look at how to use `@angular/fire` to connect an Angular application with the Firebase Emulator Suite to start prototyping your apps.
 
-There are four supported emulators, all of them available at the Firebase suite workflow:
+There are five supported emulators, all of them available at the Firebase suite workflow:
 
 - [Authentication Emulator](https://firebase.google.com/docs/emulator-suite/connect_auth)
 - [Realtime Database Emulator](https://firebase.google.com/docs/emulator-suite/connect_rtdb)
 - [Cloud Firestore Emulator](https://firebase.google.com/docs/emulator-suite/connect_firestore)
 - [Cloud Functions Emulator](https://firebase.google.com/docs/emulator-suite/connect_functions)
+- [Cloud Storage Emulator](https://firebase.google.com/docs/emulator-suite/connect_storage)
```
Comment (hmmmmniek) [inline site/src/get-started/local-development.md:18]: does not exist yet, but i expect the docs guys will use this url?

**Comment by davideast** [comment]
Hey! The storage emulator is in alpha and is not ready for AngularFire just yet.

**Comment by hmmmmniek** [comment]
> Hey! The storage emulator is in alpha and is not ready for AngularFire just yet.

yes, understandable, we can just keep this on hold until it is out of alpha :)

**Comment by shahmirn** [comment]
@davideast Storage emulator is out of alpha, and released as part of version 9.11.0 of https://github.com/firebase/firebase-tools/releases

Can we get this PR merged in, and a new release of angularfire? Thanks!

File: site/src/get-started/local-development.md (hunk 1 above)
Comment (shahmirn) [reply site/src/get-started/local-development.md:18]: This URL exists now.

**Comment by jamesdaniels** [comment]
I'll get this into the next minor.

**Review by aurelien-ldp — APPROVED** [review]

**Comment by johngrimsey** [comment]
Keen for this!

**Comment by jamesdaniels** [comment]
Addressed in 7.0 with `@angular/fire/compat/storage#USE_EMULATOR`

**Pull request closed without merging**